### PR TITLE
Basic setup for sh with logging function library

### PR DESCRIPTION
**Pull request by mitchty**
### Summary and Scope

Adds shellspec helpers and a basic logging library and tests for it that will be used throughout the library later.

Also adds github action that runs shellspec, checkbashims, and shfmt so we can keep all our shell in line at pr time. Also introduces an .editorconfig so editors know the formatting conventions.

I will setup an action for running shellspec unit tests later but for now its a makefile target.

- Fixes:
- Requires:
- Relates to:

#### Issue Type

Setting up the basics of the shell side of things. We can change the layout if we want but given all this is is shell the top level seemed fine.

- Bugfix Pull Request
- Docs Pull Request
- RFE Pull Request

### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [ ] I tested this on internal system (if yes, please include results or a description of the test)
- [ ] I tested this on a vshasta system (if yes, please include results or a description of the test)
 
### Idempotency
 
n/a
 
### Risks and Mitigations
 
No real risk yet, regardless this is the simple bit of writing logging functions. The crazy stuff will come later that can/will (ab)use this.

-->

### CODE REVIEW

**Comment by mitchty** [comment]
Heh cool so it works but sh-checkers finding stuff I didn't expect, will see how to have it ignore the spec dir cause shfmt will just ruin those files right now.

**Comment by mitchty** [comment]
Ok need to figure out a better way to ignore that spec stuff until I can get shfmt to treat it ok, so removed the look at changed files part of the action.

File: .shellspec [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--require spec_helper
+
+## Default kcov (coverage) options
+# --kcov-options "--include-path=. --path-strip-level=1"
```
Comment (rustydb) [inline .shellspec:4]: So these kcov options are all placeholders since they're commented out?

**Review by rustydb — APPROVED** [review]

File: .shellspec (hunk 1 above)
Comment (mitchty) [reply .shellspec:4]: Its default from shellspec --init along with spec/spec_helper.sh. Only useful if we want to include coverage which we can at some point. For now I just used all the defaults.

**Pull request merged**